### PR TITLE
Don't use `quansight/conda-store` docker image

### PR DESCRIPTION
**Pull request by soapy1**
Fixes https://github.com/conda-incubator/conda-store/issues/626

## Description

As we prepare to consolidate the `conda-store` and `conda-store-server` packages (as per https://github.com/conda-incubator/conda-store/issues/911) we'll need to move away from using artifacts generated from the `conda-store` release. 

This PR updates:
* the docker examples to not use the `conda-store` docker image
* fixes the docker examples so that users can log in with jpuyter oauth and create environments

## Pull request checklist

- [x] Did you test this change locally?
- [x] Did you update the documentation (if required)?
- [x] Did you add/update relevant tests for this change (if required)?

### CODE REVIEW

File: examples/docker-without-nfs/assets/conda_store_config.py [hunk 1]
```diff
@@ -57,11 +58,12 @@
 #         auth settings
 # ==================================
 c.CondaStoreServer.authentication_class = JupyterHubOAuthAuthentication
-c.JupyterHubOAuthAuthentication.jupyterhub_url = "https://conda-store.localhost"
+c.JupyterHubOAuthAuthentication.jupyterhub_url = "http://conda-store.localhost"
```
Comment (peytondmurray) [inline examples/docker-without-nfs/assets/conda_store_config.py:61]: I don't have the context on why this change is needed. Could you provide some insight about this?

File: examples/docker/docker-compose.yaml [hunk 2]
```diff
@@ -142,21 +137,6 @@ services:
...
-      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
-      interval: 10s
-      timeout: 5s
-      retries: 5
-    environment:
-      MYSQL_ROOT_PASSWORD: hbgY8bkoWPMM7EjtcD6wrlVX
-      MYSQL_USER: admin
-      MYSQL_PASSWORD: password
-      MYSQL_DATABASE: conda-store
-
```
Comment (peytondmurray) [inline examples/docker/docker-compose.yaml:159]: This was unused - thanks for the cleanup!

File: examples/docker/docker-compose.yaml [hunk 3]
```diff
@@ -82,11 +74,14 @@ services:
       - "8080:8080"
 
   jupyterhub:
-    image: quansight/conda-store
+    build:
+      context: dockerfiles
+      dockerfile: jupyterhub.Dockerfile
+      args:
+        python_version: 3.11
```
Comment (peytondmurray) [inline examples/docker/docker-compose.yaml:81]: Not sure if this is actually something we need to do here (or if there's some reason to be using 3.11), but I vote to target the latest stable version if possible (3.13)

File: examples/docker-without-nfs/docker-compose.yaml [hunk 4]
```diff
@@ -1,37 +1,29 @@
...
       - "443:443"
-      - "8080:8080"
       - "9080:9080"
     volumes:
       - "/var/run/docker.sock:/var/run/docker.sock:ro"
 
   conda-store-worker:
     image: quansight/conda-store-server
-    user: 1000:1000
+    user: 1000:100
```
Comment (peytondmurray) [inline examples/docker-without-nfs/docker-compose.yaml:25]: Is this intended? If so, any context around this would be helpful.

Comment (soapy1) [reply examples/docker-without-nfs/docker-compose.yaml:25]: This user needs to match the conda store settings `c.CondaStore.default_gid` and `c.CondaStore.default_uid`. In the [conda_store_config.py ](https://github.com/conda-incubator/conda-store/pull/1004/files?diff=unified&w=0#diff-7833f5147a649457fff55e9e16579d23960205f7d475de30802c436467bec33dR24) for this example, the user gid is 100.

File: examples/docker/docker-compose.yaml (hunk 3 above)
Comment (soapy1) [reply examples/docker/docker-compose.yaml:81]: I get a conda error when running python 3.13
```
> [jupyterhub 3/7] RUN mamba create --name conda-store     python=3.13 nb_conda_kernels nodejs=18 yarn constructor conda-store     --channel conda-forge -y &&     mamba clean --all -y &&     conda clean --force-pkgs-dirs:                     
14.27                                                                            
14.27 Looking for: ['python=3.13', 'nb_conda_kernels', 'nodejs=18', 'yarn', 'constructor', 'conda-store']
14.27 
14.27 Could not solve for environment specs
14.27 The following packages are incompatible
14.27 ├─ constructor is installable with the potential options
14.27 │  ├─ constructor [1.3.0|1.3.1|...|3.0.1] would require
14.27 │  │  └─ python [2.7* |>=2.7,<2.8.0a0 ], which can be installed;
14.27 │  ├─ constructor [1.3.0|1.3.1|1.3.2|1.3.3|1.3.4] would require
14.27 │  │  └─ python 3.4* , which can be installed;
14.27 │  ├─ constructor [1.3.0|1.3.1|...|2.2.0] would require
14.27 │  │  └─ python [3.5* |>=3.5,<3.6.0a0 ], which can be installed;
14.27 │  ├─ constructor [1.5.5|2.0.0|2.0.1|2.0.2|2.0.3] would require
14.27 │  │  └─ python 3.6* , which can be installed;
14.27 │  ├─ constructor [2.1.0|2.2.0|...|3.2.1] would require
14.27 │  │  └─ python >=3.6,<3.7.0a0 , which can be installed;
14.27 │  ├─ constructor [2.2.0|2.3.0|...|3.3.1] would require
14.27 │  │  └─ python >=3.7,<3.8.0a0 , which can be installed;
14.27 │  ├─ constructor [3.0.2|3.1.0|...|3.3.1] would require
14.27 │  │  └─ python >=3.8,<3.9.0a0 , which can be installed;
14.27 │  ├─ constructor [3.1.0|3.2.0|3.2.1] would require
14.27 │  │  └─ python_abi 3.6 *_pypy36_pp73, which requires
14.27 │  │     └─ python 3.6.* *_73_pypy, which can be installed;
14.27 │  ├─ constructor [3.2.0|3.2.1|3.2.2|3.3.1] would require
14.27 │  │  └─ python >=3.9,<3.10.0a0 , which can be installed;
14.27 │  ├─ constructor [3.2.2|3.3.1] would require
14.27 │  │  └─ python >=3.10,<3.11.0a0 , which can be installed;
14.27 │  ├─ constructor [3.10.0|3.11.0|...|3.9.3] would require
14.27 │  │  └─ __win, which is missing on the system;
14.27 │  ├─ constructor [3.3.1|3.4.1|3.4.2|3.4.3] would require
14.27 │  │  └─ conda [>=4.6,<23.1.0 |>=4.6,<23.1.0a0 ] with the potential options
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.14.0] would require
14.27 │  │     │  └─ python >=3.10,<3.11.0a0 , which can be installed;
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|24.9.2] would require
14.27 │  │     │  └─ python >=3.11,<3.12.0a0 , which can be installed;
14.27 │  │     ├─ conda [22.9.0|4.10.0|...|4.9.2] would require
14.27 │  │     │  └─ python >=3.7,<3.8.0a0 , which can be installed;
14.27 │  │     ├─ conda [22.11.1|22.9.0|4.14.0] would require
14.27 │  │     │  └─ python_abi 3.8 *_pypy38_pp73, which requires
14.27 │  │     │     └─ python 3.8.* *_73_pypy, which can be installed;
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.9.2] would require
14.27 │  │     │  └─ python >=3.8,<3.9.0a0 , which can be installed;
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.14.0] would require
14.27 │  │     │  ├─ pypy3.9 [>=7.3.13 |>=7.3.15 ], which requires
14.27 │  │     │  │  └─ python 3.9.* *_73_pypy, which can be installed;
14.27 │  │     │  └─ python_abi 3.9 *_pypy39_pp73, which requires
14.27 │  │     │     └─ python 3.9.* *_73_pypy, which can be installed;
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.9.2] would require
14.27 │  │     │  └─ python >=3.9,<3.10.0a0 , which can be installed;
14.27 │  │     ├─ conda [4.10.0|4.10.1|...|4.9.2] would require
14.27 │  │     │  └─ python >=3.6,<3.7.0a0 , which can be installed;
14.27 │  │     ├─ conda [4.10.0|4.10.1|...|4.9.2] would require
14.27 │  │     │  └─ python_abi 3.6 *_pypy36_pp73, which can be installed (as previously explained);
14.27 │  │     ├─ conda [4.10.1|4.10.2|...|4.13.0] would require
14.27 │  │     │  └─ python_abi 3.7 *_pypy37_pp73, which requires
14.27 │  │     │     └─ python 3.7.* *_73_pypy, which can be installed;
14.27 │  │     └─ conda [4.6.1|4.6.10|...|4.8.3] would require
14.27 │  │        └─ python >=2.7,<2.8.0a0 , which can be installed;
14.27 │  ├─ constructor [3.10.0|3.11.0|...|3.9.3] would require
14.27 │  │  └─ conda >=4.6  with the potential options
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.14.0], which can be installed (as previously explained);
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|24.9.2], which can be installed (as previously explained);
14.27 │  │     ├─ conda [22.9.0|4.10.0|...|4.9.2], which can be installed (as previously explained);
14.27 │  │     ├─ conda [22.11.1|22.9.0|4.14.0], which can be installed (as previously explained);
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.9.2], which can be installed (as previously explained);
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.14.0], which can be installed (as previously explained);
14.27 │  │     ├─ conda [22.11.1|22.9.0|...|4.9.2], which can be installed (as previously explained);
14.27 │  │     ├─ conda [4.10.0|4.10.1|...|4.9.2], which can be installed (as previously explained);
14.27 │  │     ├─ conda [4.10.0|4.10.1|...|4.9.2], which can be installed (as previously explained);
14.27 │  │     ├─ conda [4.10.1|4.10.2|...|4.13.0], which can be installed (as previously explained);
14.27 │  │     ├─ conda [4.6.1|4.6.10|...|4.8.3], which can be installed (as previously explained);
14.27 │  │     ├─ conda [23.1.0|23.3.0|...|23.7.3] would require
14.27 │  │     │  └─ pypy3.8 >=7.3.11  with the potential options
14.27 │  │     │     ├─ pypy3.8 7.3.11 would require
14.27 │  │     │     │  └─ python 3.8.* *_73_pypy, which can be installed;
14.27 │  │     │     └─ pypy3.8 7.3.11 would require
14.27 │  │     │        └─ python 3.8.* *_7.3_pypy, which can be installed;
14.27 │  │     ├─ conda [23.1.0|23.3.0|...|23.7.4] would require
14.27 │  │     │  └─ pypy3.9 [>=7.3.11 |>=7.3.12 ] with the potential options
14.27 │  │     │     ├─ pypy3.9 [7.3.11|7.3.12|7.3.13|7.3.15|7.3.16], which can be installed (as previously explained);
14.27 │  │     │     └─ pypy3.9 7.3.12 would require
14.27 │  │     │        └─ python 3.9.* *_7.3_pypy, which can be installed;
14.27 │  │     └─ conda [23.10.0|23.11.0|...|24.9.2] would require
14.27 │  │        └─ python_abi 3.12.* *_cp312, which requires
14.27 │  │           └─ python 3.12.* *_cpython, which can be installed;
14.27 │  └─ constructor [3.10.0|3.11.0|3.9.3] would require
14.27 │     └─ __osx, which is missing on the system;
14.27 └─ python 3.13**  is not installable because there are no viable options
14.27    ├─ python [3.13.0|3.13.1] conflicts with any installable versions previously reported;
14.27    └─ python [3.13.0rc1|3.13.0rc2|3.13.0rc3] would require
14.27       └─ _python_rc, which does not exist (perhaps a missing channel).
------
```

Ok to settle for 3.12?

Comment (peytondmurray) [reply examples/docker/docker-compose.yaml]: Oh right, last I checked conda hasn't been packaged for 3.13 yet :/

Let's do 3.12.
